### PR TITLE
Well I made one too...

### PR DESCRIPTION
**Pull request by demipixel**

### CODE REVIEW

**Comment by rom1504** [comment]
I think it would be nice to compare and then PR the implementation to https://github.com/109C/minecraft-chat-parser/blob/master/index.js and use it ;)

**Comment by 109C** [comment]
And fix the checks failing, not crashing is an important feature...

**Comment by rom1504** [comment]
@109C nah the failing test is because he commit @demipixel based his PR on had a little bug in the test, nothing wrong.

**Comment by rom1504** [comment]
The produced json is not vanilla. So mineflayer can't parse it.

Way to fix that : 
1. get a vanilla example by using nmp/mineflayer
2. make it fit
3. make sure the result is correct both in the game and that mineflayer can parse it

**Comment by rom1504** [comment]
Example:
```json
{
  "translate": "chat.type.text",
  "with": [
    {
      "insertion": "rom1504",
      "clickEvent": {
        "action": "suggest_command",
        "value": "/msg rom1504 "
      },
      "hoverEvent": {
        "action": "show_entity",
        "value": "{name:\"rom1504\",id:\"25827243-f187-3849-bcbc-8f1798bea6bb\"}"
      },
      "text": "rom1504"
    },
    "test"
  ]
}
```

**Comment by rom1504** [comment]
Important part:
```json
{
  "translate": "chat.type.text",
  "with": [
    {
      "insertion": "rom1504",
      "text": "rom1504"
    },
    "test"
  ]
}
```

**Comment by rom1504** [comment]
a better description of the chat format http://minecraft.gamepedia.com/Commands#Raw_JSON_text

File: src/lib/plugins/chat.js [hunk 1]
```diff
@@ -1,10 +1,122 @@
...
+    function createJSON() {
+      if (!text.trim()) return;
+      messageList.push({
+        text: text,
+        color: color,
+        bold: bold,
+        italic: italic,
+        underlined: underlined,
+        strikethrough: strikethrough,
+        obfuscated: random
```
Comment (rom1504) [inline src/lib/plugins/chat.js]: fyi, `{text,color,bold,italic,underlined,strikethrough,obfuscated}` works in es6

**Comment by rom1504** [comment]
merged, see aeb5f888df81663d8a91891e8cce9eddd45bbee7

The remaining fixes will be tracked in https://github.com/mhsjlw/flying-squid/issues/167

**Pull request closed without merging**